test/ex_force_test.exs

```diff
@@ -613,6 +800,155 @@ defmodule ExForceTest do
+                  "errors" => [
+                    %{
+                      "statusCode" => "MALFORMED_ID",
+                      "message" => "Contact ID: id value of incorrect type: foo",
+                      "fields" => ["Id"]
+                    }
+                  ]
+                }
+              ]}
+  end
+
+  test "update_sobjects/3 - allOrNone rollback failure", %{bypass: bypass, client: client} do
+    Bypass.expect_once(bypass, "PATCH", "/services/data/v40.0/composite/sobjects", fn conn ->
+      conn
+      |> assert_json_body(%{
+        "allOrNone" => false,
+        "records" => [
+          %{
+            "attributes" => %{"type" => "Account"},
+            "email" => "myemail@email.com",
+            "id" => "001D000000IqhSLIAZ"
+          },
+          %{
+            "attributes" => %{"type" => "Account"},
+            "email" => "fooemail@email.com",
+            "id" => "foo"
+          }
+        ]
+      })
+      |> Conn.put_resp_content_type("application/json")
+      |> Conn.resp(200, """
+        [
+          {
+            "id": "001D000000IqhSLIAZ",
+            "success": false,
+            "errors": [
+              {
+                "statusCode" : "ALL_OR_NONE_OPERATION_ROLLED_BACK",
+                "message" : "Record rolled back because not all records were valid and the request was using AllOrNone header",
+                "fields" : []
+              }
+            ]
+          },
+          {
+            "success": false,
+            "errors": [
+              {
+                "statusCode": "MALFORMED_ID",
+                "message": "Contact ID: id value of incorrect type: foo",
+                "fields": [
+                  "Id"
+                ]
+              }
+            ]
+          }
+        ]
+      """)
+    end)
+
+    records = [
+      %{id: "001D000000IqhSLIAZ", attributes: %{type: "Account"}, email: "myemail@email.com"},
+      %{id: "foo", attributes: %{type: "Account"}, email: "fooemail@email.com"}
+    ]
+
+    assert ExForce.update_sobjects(client, records) ==
+             {:ok,
+              [
+                %{
+                  "id" => "001D000000IqhSLIAZ",
+                  "success" => false,
+                  "errors" => [
+                    %{
+                      "statusCode" => "ALL_OR_NONE_OPERATION_ROLLED_BACK",
+                      "message" =>
+                        "Record rolled back because not all records were valid and the request was using AllOrNone header",
+                      "fields" => []
+                    }
+                  ]
+                },
+                %{
+                  "success" => false,
+                  "errors" => [
+                    %{
+                      "statusCode" => "MALFORMED_ID",
+                      "message" => "Contact ID: id value of incorrect type: foo",
+                      "fields" => ["Id"]
+                    }
+                  ]
+                }
+              ]}
```

Comment:
the scenario where:
- the API request was well-formed
- the request was executed
- the request failed to a data condition
- every operation was rolled back

leads to a 200 from Salesforce and every record operation `success: false`

this is unintuitive could conceivably be returned as an `{:error, ...}` but I figured `{:ok, ...}` better reflected how Salesforce responded